src/Common/Thermodynamics/relations.jl

```diff
@@ -1353,6 +1378,96 @@ function saturation_adjustment(
     end
 end
 
+"""
+    saturation_adjustment_ρpq(
+        param_set,
+        ρ,
+        p,
+        q_tot,
+        phase_type,
+        maxiter,
+        temperature_tol
+    )
+Compute the temperature that is consistent with
+ - `param_set` an `AbstractParameterSet`, see the [`Thermodynamics`](@ref) for more details
+ - `ρ` (moist-)air density
+ - `p` pressure
+ - `q_tot` total specific humidity
+ - `phase_type` a thermodynamic state type
+ - `maxiter` maximum iterations for non-linear equation solve
+ - `temperature_tol` temperature tolerance
+by finding the root of
+
+```
+T - air_temperature_from_ideal_gas_law(
+        param_set,
+        p,
+        ρ,
+        PhasePartition_equil(param_set, T, ρ, q_tot, phase_type),
+    )
+```
+using Newtons method using ForwardDiff.
+See also [`saturation_adjustment`](@ref).
+"""
+function saturation_adjustment_ρpq(
+    param_set::APS,
+    ρ::FT,
+    p::FT,
+    q_tot::FT,
+    phase_type::Type{<:PhaseEquil},
+    maxiter::Int,
+    temperature_tol::FT = eps(FT),
+) where {FT <: Real}
+    tol = SolutionTolerance(temperature_tol)
+    q_pt = PhasePartition(q_tot)
+    T_0 = air_temperature_from_ideal_gas_law(param_set, p, ρ, q_pt)
```

Comment:
I would avoid `T_0` here. We have reserved that for the thermodynamic reference temperature. Overloading it here creates confusion. `T_init` or something similar may be better.

Reply:
Good point. Will update